app/src/main/java/com/globant/harrypotterapp/viewmodel/SpellsViewModel.kt

```diff
@@ -25,10 +31,20 @@ class SpellsViewModel(private val getSpellsFromAPIUseCase: GetSpellsFromAPIUseCa
         withContext(Dispatchers.IO) { getSpellsFromAPIUseCase.invoke() }.let { result ->
             when (result) {
                 is Result.Success -> {
+                    withContext(Dispatchers.IO) { updateSpellsDataBaseUseCase.invoke(result.data) }
```

Comment:
You should call a usecase and that usecase will check from where get the info. That logic shouldn't be in the viewmodel

Reply:
Refactored, thanks Mau by the explanation!